toolchain/base/kind_switch.h

```diff
@@ -222,8 +222,12 @@ consteval auto ForCase() -> auto {
   }
 }
 
-// Given `CARBON_KIND_SWITCH(value)` and `CARBON_KIND(CaseT name)` this
-// generates `value.As<CaseT>()`.
+// Given `CARBON_KIND_SWITCH(value)` and `CARBON_KIND(CaseT name)` this converts
+// the `value` to `CaseT`.
+//
+// For types with a `kind()` accessor this uses `As<CaseT>`.
+//
+// For `std::variant<...>` this uses  `std::get<CaseT>`.
```

Comment:
```suggestion
// For types with a `kind()` accessor this uses `value.As<CaseT>`.
//
// For `std::variant<...>` this uses `std::get<CaseT>(value)`.
```
Given that the functions are called in quite different ways I think it's useful to indicate that `As` is called as a member function.